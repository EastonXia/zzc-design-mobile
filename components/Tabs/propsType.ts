export interface TabsProps {
    prefixCls?: string,
    className?: string
    tabBarPosition?: string,
    tabs: Array<tabs>,
    style?: React.CSSProperties,
    tabBarUnderlineStyle?: React.CSSProperties,
    defaultIndex: number,
    index: number,
    onChange: Function,
    maxTabLength: number,
    swipeDirection?: 'vertical' | 'horizontal',
    animated?: boolean
    swipeable?: boolean
    isOpenTabBarScroll?: boolean
    isOpenTabContentScroll?: boolean
}

export interface TabBarProps {
    tabBarPosition?: string
    currIndex: number
    preIndex: number
    prefixCls?: string,
    tabs: Array<tabs>,
    tabBarUnderlineStyle?: React.CSSProperties,
    onChange: Function,
    maxTabLength: number,
    tabDirection?: 'vertical' | 'horizontal',
    animated?: boolean
    isOpenTabBarScroll?: boolean
}

export interface TabContentProps {
    swipeDirection?: 'vertical' | 'horizontal',
    currIndex?: number
    prefixCls?: string
    animated?: boolean
    swipeable?: boolean
    isOpenTabContentScroll?: boolean
    onChange: Function
    tabs: Array<tabs>
}

export declare type tabs = {
    title: string | JSX.Element
};