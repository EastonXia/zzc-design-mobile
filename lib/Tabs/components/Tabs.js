'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _TabBar = require('./TabBar');

var _TabBar2 = _interopRequireDefault(_TabBar);

var _TabContent = require('./TabContent');

var _TabContent2 = _interopRequireDefault(_TabContent);

var _typeof = require('../../_util/typeof');

var _config = require('../../_util/config');

var _config2 = _interopRequireDefault(_config);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Tabs = function (_React$PureComponent) {
    (0, _inherits3.default)(Tabs, _React$PureComponent);

    function Tabs(props) {
        (0, _classCallCheck3.default)(this, Tabs);

        var _this = (0, _possibleConstructorReturn3.default)(this, (Tabs.__proto__ || (0, _getPrototypeOf2.default)(Tabs)).call(this, props));

        var index = props.index ? props.index : props.defaultIndex;
        _this.state = {
            currIndex: index,
            preIndex: index
        };
        _this.onChange = _this.onChange.bind(_this);
        return _this;
    }

    (0, _createClass3.default)(Tabs, [{
        key: 'componentWillReceiveProps',
        value: function componentWillReceiveProps(nextPros) {
            if (nextPros.index != null && nextPros.index != this.state.currIndex) {
                this.setState({
                    currIndex: nextPros.index,
                    preIndex: this.state.currIndex
                });
            }
        }
    }, {
        key: 'onChange',
        value: function onChange(key) {
            if ((0, _typeof.isFunction)(this.props.onChange)) {
                this.props.onChange(key);
            }
            this.goToTab(key);
        }
    }, {
        key: 'goToTab',
        value: function goToTab(key) {
            var currIndex = this.state.currIndex;
            // 当tabs是受控时，不主动更新state，通过外部改变props来改变tabs的state

            if ((0, _typeof.isNull)(this.props.index)) {
                this.setState({
                    currIndex: key,
                    preIndex: currIndex
                });
            }
        }
    }, {
        key: 'render',
        value: function render() {
            var _props = this.props,
                prefixCls = _props.prefixCls,
                style = _props.style,
                children = _props.children,
                tabBarPosition = _props.tabBarPosition,
                className = _props.className;

            var direction = tabBarPosition == 'top' || tabBarPosition == 'bottom' ? 'horizontal' : 'vertical';
            var cls = (0, _classnames2.default)(className, prefixCls + '-box', prefixCls + '-box-' + tabBarPosition);
            return _react2.default.createElement(
                'div',
                { className: cls, style: style },
                _react2.default.createElement(_TabBar2.default, (0, _extends3.default)({}, this.props, { tabDirection: direction }, this.state, { onChange: this.onChange })),
                children && _react2.default.createElement(
                    _TabContent2.default,
                    (0, _extends3.default)({}, this.props, this.state, { onChange: this.onChange }),
                    children
                )
            );
        }
    }]);
    return Tabs;
}(_react2.default.PureComponent);

exports.default = Tabs;

Tabs.defaultProps = {
    prefixCls: _config2.default.cls + '-tabs',
    className: '',
    style: {},
    tabBarUnderlineStyle: {},
    tabs: [],
    defaultIndex: 0,
    index: null,
    onChange: function onChange() {},
    maxTabLength: 3,
    swipeDirection: 'horizontal',
    animated: true,
    swipeable: true,
    isOpenTabBarScroll: true,
    isOpenTabContentScroll: true,
    tabBarPosition: 'top'
};