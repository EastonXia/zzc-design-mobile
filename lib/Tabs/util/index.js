'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.setLayoutScroll = setLayoutScroll;
exports.getTabItemSize = getTabItemSize;
exports.getTranslate3dStyle = getTranslate3dStyle;
function setLayoutScroll(layout, offset, unit, isVertical) {
    var value = isVertical ? '0px, ' + offset + unit + ', 0px' : '' + offset + unit + ', 0px, 0px';
    var translate3d = 'translate3d(' + value + ')';
    var style = layout.style;
    setTransfrom(style, translate3d);
}
function getTabItemSize(maxTabLength) {
    return (100 / maxTabLength).toFixed(4);
}
function getTranslate3dStyle(value, direction) {
    var offset = direction === 'vertical' ? '0px, ' + value + ', 1px' : value + ', 0px, 1px';
    var translate3d = 'translate3d(' + offset + ')';
    return {
        transform: translate3d,
        WebkitTransform: translate3d,
        MozTransform: translate3d
    };
}
function setTransfrom(style, value) {
    style.transform = value;
    style.WebkitTransform = value;
    style.MozTransform = value;
}