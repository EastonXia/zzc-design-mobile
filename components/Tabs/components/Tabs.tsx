import React from 'react';
import classnames from 'classnames';
import TabBar from './tabBar';
import TabContent from './tabContent';
import { isFunction, isNull } from '../../_util/typeof';
import { TabsProps } from '../propsType';
import config from '../../_util/config';
import { zzcComponentUse } from '../../_util/gtag';

export interface TabsState {
    currIndex: number
    preIndex: number
}
export default class Tabs extends React.PureComponent<TabsProps, TabsState> {
    static defaultProps = {
        prefixCls: `${config.cls}-tabs`,
        className: '',
        style: {},
        tabBarUnderlineStyle: {},
        tabs: [],
        defaultIndex: 0,
        index: null,
        onChange: () => { },
        maxTabLength: 3,
        swipeDirection: 'horizontal',
        animated: true,
        swipeable: true,
        isOpenTabBarScroll: true,
        isOpenTabContentScroll: true,
        tabBarPosition: 'top'
    }

    constructor( props: TabsProps ) {
        super( props );
        const index = props.index ? props.index : props.defaultIndex;
        this.state = {
            currIndex: index,
            preIndex: index
        };
        this.onChange = this.onChange.bind( this );
        zzcComponentUse( 'Tabs', '组件渲染' );
    }

    componentWillReceiveProps ( nextPros: any ): void {
        if ( nextPros.index != null && nextPros.index != this.state.currIndex ) {
            this.setState( {
                currIndex: nextPros.index,
                preIndex: this.state.currIndex
            } );
        }
    }

    onChange ( key: number ): void {
        if ( isFunction( this.props.onChange ) ) {
            this.props.onChange( key );
        }
        this.goToTab( key );
    }

    goToTab ( key: number ): void {
        const { currIndex } = this.state;
        // 当tabs是受控时，不主动更新state，通过外部改变props来改变tabs的state
        if ( isNull( this.props.index ) ) {
            this.setState( {
                currIndex: key,
                preIndex: currIndex
            } );
        }
    }

    render (): JSX.Element {
        const { prefixCls, style, children, tabBarPosition, className } = this.props;
        const direction = tabBarPosition == 'top' || tabBarPosition == 'bottom' ? 'horizontal' : 'vertical';
        const cls = classnames(
            className,
            `${prefixCls}-box`,
            `${prefixCls}-box-${tabBarPosition}`
        );
        return (
            <div
                className={cls}
                style={style}
            >
                <TabBar
                    {...this.props}
                    tabDirection={direction}
                    {...this.state}
                    onChange={this.onChange}
                />
                {
                    children &&
                    <TabContent
                        {...this.props}
                        {...this.state}
                        onChange={this.onChange}
                    >
                        {children}
                    </TabContent>
                }
            </div>
        );
    }
}