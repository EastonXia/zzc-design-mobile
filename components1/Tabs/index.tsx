import React, { PureComponent } from 'react';
import Error from '../_util/Error';
import Tabs from './components/Tabs';
import { TabsProps } from './propsType';
import './index.scss';

export default class TabsEntrance extends PureComponent<TabsProps, any> {
    render() {
        return (
            <Error componentName='Tabs'>
                <Tabs {...this.props} />
            </Error>
        );
    }
}