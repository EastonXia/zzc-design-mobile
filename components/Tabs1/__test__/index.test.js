import React from 'react';
import {render} from 'enzyme';
import {renderToJson} from 'enzyme-to-json';
import Tabs from '../index.tsx';
test( 'Modal', () => {
    const wrapper = render(
        <Tabs
            maxTabLength={3}
            index={0}
            tabs={[
                { title: 'tab 1' },
                { title: 'tab 2' },
                { title: 'tab 3' },
                { title: 'tab 4' },
                { title: 'tab 5' }
            ]}
            onChange={() => {}}
        >
            <div className='item-content'>
                <p>this is 1 content</p>
            </div>
            <div className='item-content'>
                <p>this is 2 content</p>
            </div>
            <div className='item-content'>
                <p>this is 3 content</p>
            </div>
            <div className='item-content'>
                <p>this is 4 content</p>
            </div>
            <div className='item-content'>
                <p>this is 5 content</p>
            </div>
        </Tabs>
    );
    expect( renderToJson( wrapper ) ).toMatchSnapshot();
} );

