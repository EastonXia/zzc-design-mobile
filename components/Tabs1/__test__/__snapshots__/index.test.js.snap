// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal 1`] = `
<div
  class="zds-tabs-box zds-tabs-box-top"
>
  <div
    class="zds-tabs-bar-box"
  >
    <div
      class="zds-tabs-bar-wrap zds-tabs-bar-wrap-am"
      style="touch-action:pan-x pan-y;transform:translate3d(0, 0px, 1px);-webkit-transform:translate3d(0, 0px, 1px);-moz-transform:translate3d(0, 0px, 1px)"
    >
      <div
        class="zds-tabs-ls zds-tabs-ls-horizontal"
      >
        <div
          class="zds-tabs-ls-item zds-tabs-active"
          style="width:33.3333%"
        >
          tab 1
        </div>
        <div
          class="zds-tabs-ls-item"
          style="width:33.3333%"
        >
          tab 2
        </div>
        <div
          class="zds-tabs-ls-item"
          style="width:33.3333%"
        >
          tab 3
        </div>
        <div
          class="zds-tabs-ls-item"
          style="width:33.3333%"
        >
          tab 4
        </div>
        <div
          class="zds-tabs-ls-item"
          style="width:33.3333%"
        >
          tab 5
        </div>
        <div
          class="zds-tabs-ls-unline zds-tabs-ls-unline-top zds-tabs-ls-unline-am"
          style="width:33.3333%;transform:translate3d(0.0000%,0,0)"
        />
      </div>
    </div>
  </div>
  <div
    class="zds-tabs-content-box zds-tabs-content-box-horizontal"
  >
    <div
      class="zds-tabs-content-wrap zds-tabs-content-wrap-am"
      style="touch-action:pan-x pan-y;transform:translate3d(0, 0px, 1px);-webkit-transform:translate3d(0, 0px, 1px);-moz-transform:translate3d(0, 0px, 1px)"
    >
      <div
        class="zds-tabs-content"
      >
        <div
          class="zds-tabs-content-item"
        >
          <div
            class="item-content"
          >
            <p>
              this is 1 content
            </p>
          </div>
        </div>
        <div
          class="zds-tabs-content-item"
        >
          <div
            class="item-content"
          >
            <p>
              this is 2 content
            </p>
          </div>
        </div>
        <div
          class="zds-tabs-content-item"
        >
          <div
            class="item-content"
          >
            <p>
              this is 3 content
            </p>
          </div>
        </div>
        <div
          class="zds-tabs-content-item"
        >
          <div
            class="item-content"
          >
            <p>
              this is 4 content
            </p>
          </div>
        </div>
        <div
          class="zds-tabs-content-item"
        >
          <div
            class="item-content"
          >
            <p>
              this is 5 content
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
